### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by Arceliar** [yggdrasil-network/yggdrasil-go#48]
The main readme is in pretty good shape, but the rest of the documentation is in a sorry state.

Suggestions:

- [x] Document code in godoc format, to assist developers.
- [x] Document admin socket (blocked by #43).
- [x] ~~Rewrite or finish the whitepaper, to explain what the main parts are, what they do, and why, preferably including everything needed to write another implementation (e.g. wire formats).~~ The .io page is good enough for now, this can be revisited after things are stable and tested a bit better.
- [x] Some kind of user-friendly documentation, e.g. github pages (optional, but wanted, has some overlap with the goals of the whitepaper).
- [x] Update readme and other documentation (.io page, comments in default config) to reflect that we've switched to `hjson` format for the config file.

**Comment by neilalexander** [comment]
With regards to the user-friendly documentation, I created [yggdrasil.github.io](https://github.com/yggdrasil-network/yggdrasil-network.github.io) (and made some tweaks to the formatting). It is generated using the GitHub Pages-built-in-Jekyll from Markdown files.

For demonstration, I copied a couple of sections from `README.md`, but we can write some more user-friendly text and add additional pages as needed.
